src/core/services/google-maps-types.ts

```diff
@@ -191,3 +191,65 @@ export interface InfoWindowOptions {
   position?: LatLng|LatLngLiteral;
   zIndex?: number;
 }
+
+export interface Point {
+  x: number;
+  y: number;
+  equals(other: Point): boolean;
+  toString(): string;
+}
+
+export interface GoogleSymbol {
+  anchor?: Point;
+  fillColor?: string;
+  fillOpacity?: string;
+  labelOrigin?: Point;
+  path?: string;
+  rotation?: number;
+  scale?: number;
+  strokeColor?: string;
+  strokeOpacity?: number;
+  strokeWeight?: number;
+}
+
+export interface IconSequence {
+  fixedRotation?: boolean;
+  icon?: GoogleSymbol;
+  offset?: string;
+  repeat?: string;
+}
+
+export interface PolylineOptions {
+  clickable?: boolean;
+  draggable?: boolean;
+  editable?: boolean;
+  geodesic?: boolean;
+  icon?: Array<IconSequence>;
+  map?: GoogleMap;
+  path?: Array<LatLng>|Array<LatLng|LatLngLiteral>;
+  strokeColor?: string;
+  strokeOpacity?: number;
+  strokeWeight?: number;
+  visible?: boolean;
+  zIndex?: number;
+}
+
+export interface Polyline extends MVCObject {
+  getDraggable(): boolean;
+  getEditable(): boolean;
+  getMap(): GoogleMap;
+  getPath(): Array<LatLng>;
+  getVisible(): boolean;
+  setDraggable(draggable: boolean): void;
+  setEditable(editable: boolean): void;
+  setMap(map: GoogleMap): void;
+  setOptions(options: any): void;
+  setPath(path: Array<LatLng|LatLngLiteral>): void;
+  setVisible(visible: boolean): void;
+}
+
+export interface PolyMouseEvent extends MouseEvent {
```

Comment:
PolyMouseEvent must be exported here: https://github.com/SebastianM/angular2-google-maps/blob/master/src/core/map-types.ts#L4

Please also add a comment (because it's part of the public API)